# Deep learning based model for regulatory epigenome
I am doing my master thesis about learning DNA sequence activity like accessibility. There's a lot of research out there on this topic. Inspired by a great collection of research papers at inspired by [papers-for-molecular-design-using-DL](https://github.com/AspirinCode/papers-for-molecular-design-using-DL/blob/main/README.md?plain=1) and [Machine-learning-for-proteins](https://github.com/yangkky/Machine-learning-for-proteins/tree/master), I decided to create my own collection. My aim is to understand these papers better and share what I learn, hope this can help people who interested in this field.

**Updating ...** 

## Menu
[Reviews](#reviews)  
[Predicting chromatin accessibility from sequence](#predicting-chromatin-accessibility-from-sequence)
[Predicting gene expression from sequence](#predicting-gene-expression-from-sequence)
[Genomic Foundation Models](#genomic-foundation-models)
[DL-based enhancer design](#dl-based-enhancer-design)



## Reviews

## Predicting chromatin accessibility from sequence

## Predicting gene expression from sequence

## Genomic Foundation Models

## DL-based enhancer design
